### PR TITLE
feat(cdp): add knock hog function

### PR DESCRIPTION
**Pull request by MarconLP**
## Changes

- Adds knock hog function

## How did you test this code?

added tests

### CODE REVIEW

File: posthog/cdp/templates/knock/template_knock.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+}
+
+let body := {
+	'type': 'track',
+	'event': event.name,
+	'userId': inputs.userId,
+	'properties': inputs.include_all_properties ? event.properties : {},
+	'messageId': event.uuid,
+    'timestamp': event.timestamp
+}
```
Comment (mariusandra) [inline posthog/cdp/templates/knock/template_knock.py:22]: Tabs vs spaces... Probably best to be consistent and stick to tabs.

**Review by mariusandra — APPROVED** [review]

File: posthog/cdp/templates/knock/template_knock.py (hunk 1 above)
Comment (MarconLP) [reply posthog/cdp/templates/knock/template_knock.py]: So I've been using 4 space indents everywhere including the hog code. (I adjusted this code as well)

In the [Salesforce template](https://github.com/PostHog/posthog/blob/master/posthog/cdp/templates/salesforce/template_salesforce.py) we are using 4 space indents in the python code but only 2 space indents in the hog code.

Should we have different indents for hog vs python code?

Comment (mariusandra) [reply posthog/cdp/templates/knock/template_knock.py]: 4 spaces is fine, but you used tabs there, which give different spaces depending on who's looking at it.

Comment (MarconLP) [reply posthog/cdp/templates/knock/template_knock.py]: Perfect - I've replaced the tabs with spaces 👍

**Pull request merged**